sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Helpers/RegistryArtifact.cs

```diff
@@ -0,0 +1,418 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Containers.ContainerRegistry
+{
+    /// <summary> A helper class that groups information and operations about an image or artifact in this container registry. </summary>
+    public partial class RegistryArtifact
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly ContainerRegistryRestClient _restClient;
```

Comment:
Similar to above, this should be "*Client" and have public constructors.

Reply:
See comment above.